src/pdag/_export/dot.py

```diff
@@ -73,20 +73,36 @@ def export_dot(core_model: CoreModel, path: Path) -> None:  # noqa: C901, PLR091
 
     for name, collection in core_model.collections.items():
         graph.add_node(_mapping_node(collection))
-        key, first_item = next(iter(collection.items()))
-        if isinstance(first_item, RelationshipABC):
-            for input_ref in first_item.iter_input_refs():
-                if isinstance(input_ref, ExecInfo):
-                    if input_ref.attribute not in added_exec_infos:
-                        graph.add_node(pydot.Node(input_ref.attribute, shape="diamond"))
-                        added_exec_infos.add(input_ref.attribute)
-                    graph.add_edge(pydot.Edge(input_ref.attribute, name))
-                else:
-                    style = "dashed" if input_ref.previous else "solid"
-                    graph.add_edge(pydot.Edge(input_ref.name, name, style=style))
-
-            for output_ref in first_item.iter_output_refs():
-                style = "dashed" if output_ref.next else "solid"
-                graph.add_edge(pydot.Edge(name, output_ref.name, style=style))
+
+        input_params: set[tuple[str, str, bool]] = set()
+        output_params: set[tuple[str, str, bool]] = set()
+
+        for item in collection.values():
+            if isinstance(item, RelationshipABC):
+                for input_ref in item.iter_input_refs():
+                    if isinstance(input_ref, ExecInfo):
+                        input_params.add(("execinfo", input_ref.attribute, False))
+                    else:
+                        input_params.add(("param", input_ref.name, input_ref.previous))
+
+                for output_ref in item.iter_output_refs():
+                    output_params.add(("param", output_ref.name, output_ref.next))
+
+        for i in input_params:
+            if i[0] == "execinfo":
+                attr = i[1]
+                if attr not in added_exec_infos:
+                    graph.add_node(pydot.Node(attr, shape="diamond"))
+                    added_exec_infos.add(attr)
+                graph.add_edge(pydot.Edge(attr, name))
+            else:
+                param = i[1]
+                style = "dashed" if i[2] else "solid"
+                graph.add_edge(pydot.Edge(param, name, style=style))
```

Comment:
This is redundant, but I prefer it because checking `i[0] == "param"` explicitly conveys that the condition is intended to be _if it's `"param"`_, not _if it's not `"execinfo"`_.
```suggestion
            elif i[0] == "param":
                param = i[1]
                style = "dashed" if i[2] else "solid"
                graph.add_edge(pydot.Edge(param, name, style=style))
            else:
                msg = f"Unknown type: {i[0]}"
                raise ValueError(msg)
```